enacts/calc.py

```diff
@@ -121,6 +126,126 @@ def synthesize_enacts(variable, time_res):
     var_name = variable
     return xrds[var_name]
 
+
+def sql2GeoJSON(shapes_sql, db_sql):
+    """ Form a GeoJSON dict from sql request to a database
+
+    Parameters
+    ----------
+    shapes_sql: str
+        sql request
+    db_sql: dict
+        dictionary with host, port, user and dbname information
+    
+    Returns
+    -------
+    features: dict
+        dictionary with features as key and GeoJSON of shapes_sql as value
+
+    See Also
+    --------
+    sql2geom, geom2GeoJSON
+
+    Examples
+    --------
+    shapes_sql: select id_1 as key, name_1 as label,
+        ST_AsBinary(the_geom) as the_geom from sen_adm1
+    db_sql:
+        host: postgres
+        port: 5432
+        user: ingrid
+        dbname: iridb
+
+    Notes
+    -----
+    Description of Return is approximative as I don't know exactly what the output is
+    """
+    return geom2GeoJSON(sql2geom(shapes_sql, db_sql))
+
+
+def geom2GeoJSON(df):
+    """ Form a GeoJSON dict from a geometric object
+
+    Parameters
+    ----------
+    df: geometric object
+        shapely geometric object
+    
+    Returns
+    -------
+    features: dict
+        dictionary with features as key and GeoJSON of `geom` as value
+
+    See Also
+    --------
+    sql2geom, shapely.MultiPolygon, shapely.geometry.mapping
+
+    Notes
+    -----
+    Description of Return is approximative as I don't know exactly what the output is
+    """
+    df["the_geom"] = df["the_geom"].apply(
+        lambda x: x if isinstance(x, MultiPolygon) else MultiPolygon([x])
+    )
+    shapes = df["the_geom"].apply(shapely.geometry.mapping)
+    for i in df.index: #this adds the district layer as a label in the dict
+        shapes[i]['label'] = df['label'][i]
+    return {"features": shapes}
+
+
+def sql2geom(shapes_sql, db_sql):
+    """ Form a geometric object from sql query to a database
+
+    Parameters
+    ----------
+    shapes_sql: str
+        sql query
+    db_sql: dict
+        dictionary with host, port, user and dbname information
+    
+    Returns
+    -------
+    geometric object
+        shapely geometric object
```

Comment:
No it isn't, it's a pandas dataframe (hence the name `df`), one column of which contains shapely objects.

Incidentally, it's a plain old pandas dataframe, not a [geopandas](https://geopandas.org/en/stable/) dataframe. I remember suggesting to Drew that geopandas might be more appropriate, and that we eventually decided not to switch, but I don't have a clear memory of  the reason. It might have been that it just didn't seem like it helped enough to be worth adding another dependency to the conda environment.